tests/sentry/integrations/jira_server/test_integration.py

```diff
@@ -1,433 +1,834 @@
 
-    @responses.activate
-    def test_authentication_request_token_fails(self):
-        responses.add(
-            responses.POST,
-            "https://jira.example.com/plugins/servlet/oauth/request-token",
-            status=503,
+        external_issue = ExternalIssue.objects.create(
+            organization_id=self.organization.id, integration_id=integration.id, key="SEN-5"
         )
 
-        # Start pipeline and go to setup page.
-        self.client.get(self.setup_path)
+        IntegrationExternalProject.objects.create(
+            external_id="10100",
+            organization_integration_id=OrganizationIntegration.objects.get(
+                organization_id=self.organization.id, integration_id=integration.id
+            ).id,
+            resolved_status="10101",
+            unresolved_status="3",
+        )
 
-        # Submit credentials
-        data = {
-            "url": "https://jira.example.com/",
-            "verify_ssl": False,
-            "consumer_key": "sentry-bot",
-            "private_key": EXAMPLE_PRIVATE_KEY,
-        }
-        resp = self.client.post(self.setup_path, data=data)
-        assert resp.status_code == 200
-        self.assertContains(resp, "Setup Error")
-        self.assertContains(resp, "request token from Jira")
+        installation = integration.get_installation(self.organization.id)
+
+        with mock.patch.object(StubJiraApiClient, "transition_issue") as mock_transition_issue:
+            with mock.patch.object(installation, "get_client", get_client):
+                # test unresolve -- 21 is "in progress" transition id
+                installation.sync_status_outbound(external_issue, False, self.project.id)
+                mock_transition_issue.assert_called_with("SEN-5", "21")
+
+                # test resolve -- 31 is "done" transition id
+                installation.sync_status_outbound(external_issue, True, self.project.id)
+                mock_transition_issue.assert_called_with("SEN-5", "31")
 
     @responses.activate
-    def test_authentication_request_token_redirect(self):
+    def test_sync_assignee_outbound_case_insensitive(self):
+        self.user = self.create_user(email="bob@example.com")
+        issue_id = "APP-123"
+        assign_issue_url = "https://jira.example.org/rest/api/2/issue/%s/assignee" % issue_id
+        external_issue = ExternalIssue.objects.create(
+            organization_id=self.organization.id,
+            integration_id=self.installation.model.id,
+            key=issue_id,
+        )
         responses.add(
-            responses.POST,
-            "https://jira.example.com/plugins/servlet/oauth/request-token",
-            status=200,
-            content_type="text/plain",
-            body="oauth_token=abc123&oauth_token_secret=def456",
+            responses.GET,
+            "https://jira.example.org/rest/api/2/user/assignable/search",
+            json=[{"name": "Alive Tofu", "emailAddress": "Bob@example.com"}],
```

Comment:
This differs very slightly from the same test in Jira Cloud, since you need to pass the `name` instead of the `accountId`.

Reply:
`deadbeef123` 👉 `Alive Tofu`

ur a legend